src/extension.ts

```diff
@@ -42,11 +46,57 @@ export interface Api {
 /**
  * Activate the extension. This is the main entry point.
  */
-export async function activate(context: vscode.ExtensionContext): Promise<Api> {
+export async function activate(context: vscode.ExtensionContext): Promise<Api | undefined> {
     try {
         console.debug("Activating Swift for Visual Studio Code...");
 
-        const workspaceContext = await WorkspaceContext.create();
+        const outputChannel = new SwiftOutputChannel();
+        const toolchain: SwiftToolchain | undefined = await SwiftToolchain.create()
+            .then(toolchain => {
+                outputChannel.log(toolchain.swiftVersionString);
+                toolchain.logDiagnostics(outputChannel);
+                contextKeys.createNewProjectAvailable = toolchain.swiftVersion.isGreaterThanOrEqual(
+                    new Version(5, 8, 0)
+                );
+                return toolchain;
+            })
+            .catch(error => {
+                outputChannel.log("Failed to discover Swift toolchain");
+                outputChannel.log(error);
+                contextKeys.createNewProjectAvailable = true;
+                return undefined;
+            });
+        const workspaceContext = toolchain
+            ? await WorkspaceContext.create(outputChannel, toolchain)
+            : undefined;
+        context.subscriptions.push(...commands.register(workspaceContext));
+
+        if (!toolchain && vscode.workspace.workspaceFolders) {
```

Comment:
Why `&& vscode.workspace.workspaceFolders` someone can create a single swift file and still expect swift support.